src/main/java/com/autotune/analyzer/deployment/AutotuneDeployment.java

```diff
@@ -601,30 +630,75 @@ private static void addLayerInfo(AutotuneConfig layer) {
 					for (Container container : containers) {
 						String containerImageName = container.getImage();
 						// Check if the container image is already present in the applicationServiceStackMap, it not, add it
-						for (String autotuneObjectKey : applicationServiceStackMap.keySet()) {
-							if (applicationServiceStackMap.get(autotuneObjectKey).containsKey(containerImageName)) {
-								addLayerInfoToApplication(applicationServiceStackMap.get(autotuneObjectKey).get(containerImageName), layer);
+						if (autotuneObject != null) {
+							if (applicationServiceStackMap.get(autotuneObject.getExperimentName()).containsKey(containerImageName)) {
```

Comment:
Is this condition correct, the comment seems to contradict. Minor typo in comment change it to if

Reply:
Updated comment. However note that this code is going to completely change in my upcoming PR which adds the `ApplicationDeployment` abstraction.